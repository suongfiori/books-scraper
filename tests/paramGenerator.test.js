
const { getGenreParam } = require('../src/paramGenerator')

describe('getGenreParam', () => {
    it ('should return a map of genre parameters', async ()=> {
        const genreParams = await getGenreParam()

        expect(typeof genreParams).toBe('object')
        expect(Object.keys(genreParams).length).toBeGreaterThan(0)
        expect(genreParams).toMatchSnapshot()
    })
})


