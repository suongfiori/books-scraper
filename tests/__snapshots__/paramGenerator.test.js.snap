// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGenreParam should return a map of genre parameters 1`] = `
{
  "academic": "academic_40",
  "add a comment": "add-a-comment_18",
  "adult fiction": "adult-fiction_29",
  "art": "art_25",
  "autobiography": "autobiography_27",
  "biography": "biography_36",
  "business": "business_35",
  "childrens": "childrens_11",
  "christian": "christian_43",
  "christian fiction": "christian-fiction_34",
  "classics": "classics_6",
  "contemporary": "contemporary_38",
  "crime": "crime_51",
  "cultural": "cultural_49",
  "default": "default_15",
  "erotica": "erotica_50",
  "fantasy": "fantasy_19",
  "fiction": "fiction_10",
  "food and drink": "food-and-drink_33",
  "health": "health_47",
  "historical": "historical_42",
  "historical fiction": "historical-fiction_4",
  "history": "history_32",
  "horror": "horror_31",
  "humor": "humor_30",
  "music": "music_14",
  "mystery": "mystery_3",
  "new adult": "new-adult_20",
  "nonfiction": "nonfiction_13",
  "novels": "novels_46",
  "paranormal": "paranormal_24",
  "parenting": "parenting_28",
  "philosophy": "philosophy_7",
  "poetry": "poetry_23",
  "politics": "politics_48",
  "psychology": "psychology_26",
  "religion": "religion_12",
  "romance": "romance_8",
  "science": "science_22",
  "science fiction": "science-fiction_16",
  "self help": "self-help_41",
  "sequential art": "sequential-art_5",
  "short stories": "short-stories_45",
  "spirituality": "spirituality_39",
  "sports and games": "sports-and-games_17",
  "suspense": "suspense_44",
  "thriller": "thriller_37",
  "travel": "travel_2",
  "womens fiction": "womens-fiction_9",
  "young adult": "young-adult_21",
}
`;
